Core/Src/ds3231.c

```diff
@@ -1,53 +1,104 @@
 #include "ds3231.h"
 
-void DS3231_GetDate(DS3231_t* pDS)
+uint32_t DS3231_GetDate(DS3231_t* pDS)
 {
+  if(pDS == NULL)
```

Comment:
It would be better to use `assert` here. Because if `pDS == NULL` it is definitely the mistake of programmer and we must stop our firmware. You can read about asserts [here](https://interrupt.memfault.com/blog/asserts-in-embedded-systems).

Reply:
You can left the condition to return the error code if `NULL` is detected but please, add an assert for Debug build also.